lighthouse-core/gather/gatherers/scripts.js

```diff
@@ -123,15 +139,28 @@ class Scripts extends FRGatherer {
       // It's nice to display the user-provided value in Lighthouse, so we add a field 'name'
       // to make it clear this is for presentational purposes.
       // See https://chromium-review.googlesource.com/c/v8/v8/+/2317310
+      let name = event.url;
+      // embedderName is optional on the protocol because backends like Node may not set it.
+      // For our purposes, it is always set. But just in case it isn't... fallback to the url.
+      let url = event.embedderName || event.url;
+
+      // Some eval'd scripts may not have a name or url, so let's set them.
+      if (!name) name = '<eval script>';
```

Comment:
Regardless of my other comment, this condition applies to any script added dynamically, maybe we give them a more general name like `<dynamically added script>` or `<dynamic script>` or something.

Reply:
maybe `<compiled from string>`? dynamic is overloaded with the concept of dynamic script imports.